microcosm_flask/sync/push.py

```diff
@@ -47,7 +50,25 @@ def push_json(inputs, base_url):
             scheme=parsed_base_url.scheme,
             netloc=parsed_base_url.netloc,
         ))
-        push_resource_json(uri, resource)
+
+        if batch_size == 1:
+            push_resource_json(uri, resource)
+            continue
+
+        # batch handling
+        collection_uri = uri.rsplit("/", 1)[0]
+
+        if any((
+            current_uri is not None and current_uri != collection_uri,
```

Comment:
We either have an existing batch that is too large or we have a different resource (as interpreted by the collection URI). Either way, we need to submit the current batch before we go onto the next one.

In other words: we're NOT assuming the backend supports polymorphic updates with different resources in the same batch.